growth/too/tasks/growthdb_cgi.py

```diff
@@ -19,14 +21,105 @@
+            except (TypeError, KeyError, ValueError, AttributeError):
+                rr, dd = s_dict['ra'], s_dict['dec']
+            s_coords = SkyCoord(ra=rr*u.deg, dec=dd*u.deg)
+            s_dict["ra"], s_dict["dec"] = s_coords.ra, s_coords.dec
+            ra_h = '{0:02d}'.format(int(s_coords.ra.hms.h))
+            ra_m = '{0:02d}'.format(int(s_coords.ra.hms.m))
+            ra_s = '{0:02d}'.format(int(s_coords.ra.hms.s))
+            ra_sd = '{0:02d}'.format(int(100*(s_coords.ra.hms.s -
+                                     int(s_coords.ra.hms.s))))
+            s_dict["ra_string"] = f"{ra_h}:{ra_m}:{ra_s}.{ra_sd}"
+            dec_d = '{0:02d}'.format(int(s_coords.dec.dms.d))
+            dec_m = '{0:02d}'.format(abs(int(s_coords.dec.dms.m)))
+            dec_s = '{0:02d}'.format(int(abs(s_coords.dec.dms.s)))
+            dec_sd = '{0:02d}'.format(int(100 * abs(s_coords.dec.dms.s -
+                                      int(s_coords.dec.dms.s))))
+            s_dict["dec_string"] = f"{dec_d}:{dec_m}:{dec_s}.{dec_sd}"
+        else:
+            s_dict["ra_string"] = None
+            s_dict["dec_string"] = None
+
+        sources_growth_marshal.append(s_dict)
+
+    return sources_growth_marshal
+
+
+def select_sources_in_contour(sources_growth_marshal, skymap, level=90):
+    """Select only those sources within a given contour
+    level of the skymap"""
+
+    skymap_prob = skymap.flat_2d
+    sort_idx = np.argsort(skymap_prob)[::-1]
+    csm = np.empty(len(skymap_prob))
+    csm[sort_idx] = np.cumsum(skymap_prob[sort_idx])
+    ipix_keep = sort_idx[np.where(csm <= level/100.)[0]]
+    nside = hp.pixelfunc.get_nside(skymap_prob)
+    sources_growth_marshal_contour = list(s for s in sources_growth_marshal
+                                          if ("ra" in s)
+                                          and (hp.ang2pix(
+                                                          nside,
+                                                          0.5 * np.pi -
+                                                          np.deg2rad(s["dec"].
+                                                                     value),
+                                                          np.deg2rad(s["ra"].
+                                                                     value)
+                                                         ) in ipix_keep
+                                               )
+                                          )
+
+    return sources_growth_marshal_contour
+
+
 def get_programidx(program_name):
     """Given a program name, it returns the programidx"""
 
     r = requests.post(
         'http://skipper.caltech.edu:8080/cgi-bin/growth/list_programs.cgi')
     r.raise_for_status()
     programs = r.json()
-    program_dict = {p['name']: p['programidx'] for p in enumerate(programs)}
+    program_dict = {p['name']: p['programidx'] for i, p in enumerate(programs)}
```

Comment:
This index is unused. Just drop `enumerate()`.

Reply:
Fixed.